bigquery_etl/cli/query.py

```diff
@@ -2199,12 +2205,24 @@ def deploy(
         query_files = paths_matching_name_pattern(
             name, ctx.obj["TMP_DIR"], project_id, ["query.*", "script.sql"]
         )
-        if not query_files:
+        metadata_files = paths_matching_name_pattern(
+            name, ctx.obj["TMP_DIR"], project_id, ["metadata.yaml"]
+        )
+        if not query_files and not metadata_files:
             raise click.ClickException(f"No queries matching `{name}` were found.")
 
     credentials = get_credentials()
     id_token = get_id_token(credentials=credentials)
 
+    query_file_paths = [query_file.parent for query_file in query_files]
+    metadata_files_without_query_file = [
+        metadata_file
+        for metadata_file in metadata_files
+        if metadata_file.parent not in query_file_paths
+        and not (metadata_file.parent / VIEW_FILE).exists()
+        and not (metadata_file.parent / MATERIALIZED_VIEW).exists()
```

Comment:
Might need to handle udfs as well otherwise they'll raise a lot of `SkippedDeployException`

Reply:
Also `stored_procedure.sql`.  Maybe it's better to explicitly look for all metadata with no sql or query.py